### PR TITLE
introduce metadata object update

### LINKED ISSUE #175: add support for metadata object update

### PR DESCRIPTION
**Pull request by pirkogdc**
closes #175

### CODE REVIEW

**Comment by martiner** [comment]
create an issue, otherwise looks fine

**Comment by pirkogdc** [comment]
issue created, commits squashed - please check the milestone - I think 0.15 is already released

File: src/main/java/com/gooddata/md/ObjUpdateException.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public class ObjUpdateException extends GoodDataException {
+
+    /**
+     * Construct a new instance of ObjUpdateException.
+     *
+     * @param obj   the metadata object you're trying to update
+     * @param cause the cause of error
+     * @param <T>   the type of metadata object you're trying to update
+     */
+    public <T extends Obj> ObjUpdateException(T obj, Throwable cause) {
```
Comment (martiner) [inline src/main/java/com/gooddata/md/ObjUpdateException.java]: Updatable as well?

File: src/main/java/com/gooddata/md/MetadataService.java [hunk 2]
```diff
@@ -94,10 +94,28 @@ public MetadataService(RestTemplate restTemplate) {
...
+        } catch (GoodDataException | RestClientException e) {
+            throw new ObjUpdateException(obj, e);
+        }
+    }
+
+    /**
      * Remove metadata object URI
      *
      * @param obj metadata object to remove
-     * @param <T> type of the object to be returned
```
Comment (martiner) [inline src/main/java/com/gooddata/md/MetadataService.java]: unrelated, please remove from this pr

File: src/main/java/com/gooddata/md/Updatable.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.gooddata.md;
+
+/**
+ * Marker interface for metadata objects which can be updated using PUT on objects URI (see MetadataService.updateObj).
```
Comment (martiner) [inline src/main/java/com/gooddata/md/Updatable.java]: "using PUT on objects URI" is a bit misleading. Either speak with whole sentence - eg. "using HTTP PUT method..." or remove.
ad java doc link to `updateObj()`

**Pull request merged**